feat(data-orchestrator): add action tasks page for serverless functions MTA-7255 (#6637)
scaleway/docs-content: pages/data-orchestrator/reference-content/action-tasks.mdx

---
title: Workflow definition action tasks

> description: Learn the action tasks format for YAML Workflow definition files in Scaleway Data Orchestrator.

Reviewer: ```suggestion
description: Learn how to use the action-tasks format in YAML workflow definition files for Scaleway Data Orchestrator.
```

Reviewer: If it's too much, ignore. 
Then just review whether it's "Workflow definition" or "workflow definition".

> tags: data-orchestrator workflow create logic tasks

Reviewer: Should this say "action tasks" instead of "logic tasks"? I don't know because I don't know the product, just mentioning it in case it's relevant.

> tags: data-orchestrator workflow create logic tasks

Author: ```suggestion
tags: data-orchestrator workflow create action tasks
```

dates:
  posted: 2026-06-22
---

> Data Orchestrator in Code view requires the user to upload a YAML file to use as a workflow definition.

Reviewer: ```suggestion
Data Orchestrator in Code view requires that you upload a YAML file to use as a workflow definition.
```

> This page lists the different action tasks supported in Data Orchestrator workflows.

Reviewer: ```suggestion
This page lists the different action tasks supported in Data Orchestrator workflow definitions.
```

Reviewer: I am suggesting this change to somehow establish a logical link between the first sentence and this sentence.

After this, I would also add a sentence like: Action tasks allow you to blablabla....

I just feel that some context is missing. As a reader, I'd like to understand why I would want to use action tasks, what is their role, etc.

<Tabs>
  <TabsTab label="Serverless Job">
    This is a basic example of a workflow calling one Serverless Job.
      ```yaml
      document:
        dsl: 1.0.3
        namespace: <namespace_name>
        name: <workflow_name>
        version: 1.0.0
      do:
      - task:
          call: "serverless_job"
          metadata:
            description: "The first job"
          with:
            id: "<Serverless_job_ID>"
            region: "fr-par"
      ```

>     Replace the ID placeholders with the IDs of your Serverless Job. You can copy it from the [Serverless Jobs page](https://console.scaleway.com/serverless-jobs/) of the console.

Reviewer: ```suggestion
    Replace the ID placeholder with the ID of your Serverless Job. You can copy it from the [Serverless Jobs page](https://console.scaleway.com/serverless-jobs/) of the console.
```

  </TabsTab>

  <TabsTab label="Serverless Function">
    This is an example of a how to get a Serverless Function.
      ```yaml
      document:
        dsl: 1.0.3
        namespace: <namespace_name>
        name: <workflow_name>
        version: 1.0.0
      do:
        - callFunctionGet:
            call: "serverless_function"
            with:
              method: GET
              id: "<Serverless_function_ID>"
              region: fr-par
              output: "response"
        - setOutput:
            set:
              result: "${ . }"       
      ```

>     Replace the ID placeholders with the IDs of your Serverless Function. You can copy it from the [Serverless Functions page](https://console.scaleway.com/functions/) of the console.

Reviewer: ```suggestion
    Replace the ID placeholder with the ID of your Serverless Function. You can copy it from the [Serverless Functions page](https://console.scaleway.com/functions/) of the console.
```
